src/components/views/Booktrak/BooktrakListings.tsx

```diff
@@ -4,98 +4,158 @@ import React, { useState, useEffect } from "react";
+  displayBuyAndSell?: boolean;
+}) => {
   const wso = useAppSelector(getWSO);
   const navigateTo = useNavigate();
-  const [buyListings, updateBuyListings] = useState<ModelsBookListing[]>([]);
-  const [sellListings, updateSellListings] = useState<ModelsBookListing[]>([]);
+  const [listings, updateListings] = useState<ModelsBookListingWithTitle[]>([]);
+  const [buyListings, updateBuyListings] = useState<
+    ModelsBookListingWithTitle[]
+  >([]);
+  const [sellListings, updateSellListings] = useState<
+    ModelsBookListingWithTitle[]
+  >([]);
+
+  const [currentPage, updateCurrentPage] = useState(0);
+  const maxListingsPerPage = 20;
 
   useEffect(() => {
     const loadListings = async () => {
-      if (!book.id) {
-        return;
+      const params: {
+        bookID?: number;
+        listingType?: ModelsBookListing.ListingTypeEnum;
+      } = {};
+      if (book?.id) {
+        params.bookID = book.id;
       }
+      if (onlyShowBuyListings !== undefined) {
+        params.listingType = onlyShowBuyListings
+          ? ListingTypeEnum.BUY
+          : ListingTypeEnum.SELL;
+      }
+
       try {
-        const ListingTypeEnum = ModelsBookListing.ListingTypeEnum;
-        const buyListingsResponse = await wso.booktrakService.listBookListings({
-          bookID: book.id,
-          listingType: ListingTypeEnum.BUY,
-        });
-        const sellListingsResponse = await wso.booktrakService.listBookListings(
-          {
-            bookID: book.id,
-            listingType: ListingTypeEnum.SELL,
+        if (displayBuyAndSell) {
+          const buyListingsResponse =
+            await wso.booktrakService.listBookListings({
+              ...params,
+              listingType: ListingTypeEnum.BUY,
+            });
+          const sellListingsResponse =
+            await wso.booktrakService.listBookListings({
+              ...params,
+              listingType: ListingTypeEnum.SELL,
+            });
+
+          updateBuyListings(buyListingsResponse.data ?? []);
+          updateSellListings(sellListingsResponse.data ?? []);
+        } else {
+          const listingsResponse = await wso.booktrakService.listBookListings({
+            ...params,
+          });
+
+          const newListings: ModelsBookListingWithTitle[] = [];
+          for (const listing of listingsResponse.data ?? []) {
+            const bookData = await wso.booktrakService.getBook(
+              listing.bookID ?? 0
+            );
+            newListings.push({
+              ...listing,
+              bookTitle: bookData.data?.title ?? "",
+            });
           }
-        );
-        updateBuyListings(buyListingsResponse.data ?? []);
-        updateSellListings(sellListingsResponse.data ?? []);
+          updateListings(newListings ?? []);
+        }
       } catch (error) {
-        // eslint-disable-next-line @typescript-eslint/no-explicit-any
-        navigateTo("/404", { replace: true });
+        if (isServerError(error)) {
```

Comment:
Should probably still navigate to error page if there is a server error

Reply:
The reason I did this is because the useEffect gets called a few times when the page loads, and for whatever reason the first call gets a 401 unauthorized response

Reply:
You should fix the reason this happens. It's because the route isn't wrapped around a RequireScope component [here](https://github.com/WilliamsStudentsOnline/wso-react/blob/506a589bb8f09604e055bf2b0cf6d4736753f7e0/src/components/App.tsx#L182). Also the module should be lazily imported like the rest instead of directly imported [here](https://github.com/WilliamsStudentsOnline/wso-react/blob/506a589bb8f09604e055bf2b0cf6d4736753f7e0/src/components/App.tsx#L27). Make sure to add token level 3 for bookmark [here](https://github.com/WilliamsStudentsOnline/wso-react/blob/506a589bb8f09604e055bf2b0cf6d4736753f7e0/src/router-permissions.js#L14) for this to work as well.